client/my-sites/stats/site.jsx

```diff
@@ -267,23 +265,75 @@ class StatsSite extends Component {
 					</div>
 				</div>
 				<JetpackColophon />
+			</>
+		);
+	}
+
+	renderEnableStatsModule() {
+		const { slug, siteId, path } = this.props;
+		return (
+			<EmptyContent
+				illustration="/calypso/images/illustrations/illustration-404.svg"
+				title={ translate( 'Looking for Stats?' ) }
+				line={ translate(
+					'Enable Site Stats below or visit {{a}}Traffic > Site Stats{{/a}} for more options',
+					{
+						components: {
+							a: <a href={ `/marketing/traffic/${ slug }` } />,
+						},
+					}
+				) }
+				action={ translate( 'Enable Site Stats' ) }
+				actionCallback={ () => this.props.enableJetpackStatsModule( siteId, path ) }
```

Comment:
Could we declare this handler as a component method, so it won't be declared on every re-render?

Reply:
updated in https://github.com/Automattic/wp-calypso/pull/39004/commits/49193bf659b4afbf29447a5b68baa872d88c2d73